packages/remix-dev/compiler/plugins/serverNativeNodeModules.ts

```diff
@@ -0,0 +1,32 @@
+import type { Plugin } from "esbuild";
+
+// See https://github.com/evanw/esbuild/issues/1051#issuecomment-806325487
+export const serverNativeNodeModulesPlugin: Plugin = {
+  name: 'native-node-modules',
+  setup(build) {
+    // If a ".node" file is imported within a module in the "file" namespace, resolve
+    // it to an absolute path and put it into the "node-file" virtual namespace.
+    build.onResolve({ filter: /\.node$/, namespace: 'file' }, args => ({
+      path: require.resolve(args.path, { paths: [args.resolveDir] }),
```

Comment:
@jacob-ebey's comment (https://github.com/remix-run/remix/pull/3263#discussion_r910178465)
> Add `sideEffects: false` here to make sure this can be optimized out of client bundles.